src/config.rs

```diff
@@ -1889,11 +1889,13 @@ impl TiKvConfig {
         self.storage.validate()?;
 
         self.raft_store.region_split_check_diff = self.coprocessor.region_split_size / 16;
-        self.raft_store.raftdb_path = if self.raft_store.raftdb_path.is_empty() {
-            config::canonicalize_sub_path(&self.storage.data_dir, "raft")?
-        } else {
-            config::canonicalize_path(&self.raft_store.raftdb_path)?
-        };
+
+        let default_raftdb_path = config::canonicalize_sub_path(&self.storage.data_dir, "raft")?;
+        if self.raft_store.raftdb_path.is_empty() {
```

Comment:
What's wrong with original implement?

Reply:
1.`validate` may run multiple time 
2.`canonicalize_sub_path` will not create dir, and will return error if `parent` not exist
if `raftdb_path` is empty first, it will set to `canonicalize_sub_path(data_dir, "raft")`, in the next run of `validate`, `canonicalize_path("data_dir/raft")` will return error because "data_dir/raft" may not exists, beside this it have the same behavior as master branch.